### DISCUSSION TITLE
perf script to trace packet

### DISCUSSION THREAD

**Discussion by qmaldon** [janestreet/magic-trace#259] [Q&A] [answered]
Hi guys,

I've been using magic-trace for a couple of months now. Its capabilities have helped me a lot to improve performance.

Now I'm starting to capture a multiprocess, multithread service. Using something like:
perf record --pid 123,456,789 --filter="start func @ program" --filter="stop func2 @ program" -e intel_pt/cyc=1,noretcomp=1/u

I obtain a perf.data file (or a directory when we need to preserve kernel symbols I think).

The point is, **can I generate a trace packet from this perf.data ?**

I'm aware of perf script --ns --itrace=b -F pid,tid,time,flags,ip,addr,sym,symoff, but I'm not sure how to pass that output to magi-trace for its processing.

I tried magic-trace decode, but I'm unable to run it as expected:
$ magic-trace decode -executable [REDACTED] -working-directory ./work_dir
[ Decoding, this takes a while... ]
(monitor.ml.Error
 (Sys_error "./work_dir/hits.sexp: No such file or directory")
...

**Comment by qmaldon** [comment, marked as answer]
Found a workaroud by creating a **hits.sexp** file in the work dir with this content:
()

**Reply by Xyene** [reply]
Yeah, that sounds right. In some sense the hits sexp is optional, so we'd more than welcome a PR making magic-trace just ignore the `hits.sexp` being missing.
